server/channel.js

```diff
@@ -0,0 +1,72 @@
+var WebSocketServer = require("ws").Server;
+var fs              = require("fs");
+
+// this channel.js is the server-side abstraction over websockets
```

Comment:
I don't think it's worth the effort. The implementations differ quite a bit in functionality, all the if(server) {} elseif (client) will make the code hard to manage. Although I do agree that there is quite a lot of duplicate code.